### DISCUSSION TITLE
Gerenciador de Requisição/Promise.

### DISCUSSION THREAD

**Discussion by danieloprado** [eduzz/front-end#5] [Ideas]
Devemos utilizar o ReactQuery ou apenas o usePromise já resolve?

O react query tem ferramentas boas para gerenciar o state da requisição, e até ser manipulado fora do componente que ele foi criado, mas sinto que as vezes ele complica um pouco, principalmente na parte de mutation que normalmente é usado em submit do formulário:

Ex:
```tsx
// List
 const query = useQuery({
    queryKey: ['user', 'list'],
    queryFn: () => get('/admin/users', params)
 });

// Form
const mutation = useMutation({
    mutationKey: ['user', 'form'],
    mutationFn: (model: FormModel<typeof form>) => post<User>('/admin/users', model),
    onSuccess: data => {
      queryClient.refetchQueries(['user', 'list']); // refresh da lista sem precisar fazer um callback

      message.success('Usuário salvo com sucesso!');
    }
});
```

**Comment by ffernandomoraes** [comment]
Eu gosto da ideia do `React Query`, principalmente pela facilidade no gerenciamento das requests e pelo hook do `useQuery`, não precisamos ficar criando vários states para controlar o carregamento e erros da página. 

Tem toda parte de cache também que é interessante em alguns contextos. 

Eu adotaria como um padrão da empresa, mas acredito que devemos desenvolver uma POC mostrando como fazer algumas coisas, explicando outras pra todos conseguirem visualizar o real valor da lib.

**Reply by miguelaugl** [reply]
Concordo!

**Comment by danieloprado** [comment]
Perfeito.
